spec/formatting.md

```diff
@@ -721,6 +725,26 @@ These are divided into the following categories:
     > when * {The value is not one.}
     > ```
 
+  - **Reserved Syntax errors** occur when an expression uses syntax reserved for
+    future standardization,
+    or for private implementation use that is not supported by the current implementation.
+
+    > For example, attempting to format this message
+    > would always result in an Reserved Syntax error:
+    >
+    > ```
+    > {The value is {@horse}.}
+    > ```
+    >
+    > Attempting to format this message would result in an Reserved Syntax error
+    > if done within a context that does not support the `^` private use sigil:
```

Comment:
I think this might be better as:

```suggestion
> Attemping to format the following message would result in a _Reserved Syntax error_
> if done in a context that does not support the private use sigils or where no private
> use function `^private` is available:
```

Reply:
I don't think we want to refer to "private use functions" as a concept. We've been pretty good so far to not assign any meaning to reserved annotations, as they really could mean anything.

Your `_underlines_` do remind me that we ought to apply a pass of `**_definition_**` and `_use_` styling to errors.

Reply:
@eemeli ... and yeah, we need to do a doc sweep for formatting and MUSTard consistency.

Reply:
The language that I'm proposing is attempting to define this as a well-defined error in an implementation that does not attach any meaning to `^`, and it's explicitly not defining what might happen in an implementation that does attach some significance to the sigil. It could, for instance, be a custom comment format.

> I think maybe another important consideration is that we don't want interstitial processes to yak on messages that the ultimate consumer might be able to process (or might rely on), e.g. translation tools should not reject `^private`. Only the act of formatting should produce an error for reserved or private sequences (the "I don't know what this is" error)

Agreed. That's why this is classified as a Resolution error and not a Syntax error, despite the name.

Reply:
> The language that I'm proposing is attempting to define this as a well-defined error in an implementation that does not attach any meaning to ^, and it's explicitly not defining what might happen in an implementation that does attach some significance to the sigil. It could, for instance, be a custom comment format.

Understood. I think you can't say "doesn't attach _any_ meaning to `^`" because that sigil always means "private use gorp follows". "There is a private use expression here, but I don't know what it means" or "there is a private use expression here, but I don't support formatting these".

I think what I was describing (the implementation supports private use and the sigil in question, but something is wrong with the private use expression) should just be a garden variety _Resolution error_? If we weren't trying to portmanteau the reserved and private-use sigils in a single section, you'd have some repetitive text, but we can elucidate the private use requirements separately at our leisure. 

To your point about the name, I would change it from "Reserved Syntax error" to avoid the misleading term "syntax". Perhaps "reserved expression error" or (more-correct in the ABNF) "reserved annotation error"? Or, given that we want the same errors for reserved and private-use, perhaps "Unsupported expression error"?

Reply:
Renamed the error as "Unsupported Expression", as that seems suitably inclusive of both future-spec-reserved and private-reserved annotations.